leaflet-ohm-timeslider.js

```diff
@@ -673,63 +673,72 @@ L.Control.OHMTimeSlider = L.Control.extend({
                     osmfilteringclause,
                     oldfilters
                 ];
-                // console.debug([ `OHMTimeSlider: _setupDateFiltersForLayers() ArrayFilter ${layer.id}`, oldfilters, newfilters ]);
+                // console.debug([ `OHMTimeSlider: _addDateFiltersToVectorLayers() ArrayFilter ${layer.id}`, oldfilters, newfilters ]);
             }
             else {
                 // some other condition I had not expected and need to figure out
                 console.error(oldfilters);
-                throw `OHMTimeSlider: _setupDateFiltersForLayers() got unexpected filtering condition on layer ${layer.id}`;
+                throw `OHMTimeSlider: _addDateFiltersToVectorLayers() got unexpected filtering condition on layer ${layer.id}`;
             }
 
             // apply the new filter, with the placeholder "eternal features" filter now prepended
             vecmap.setFilter(layer.id, newfilters);
         });
     },
     _removeDateFiltersFromvectorLayers: function () {
-        // in _setupDateFiltersForLayers() we rewrote the layers' filters to support date filtering, but we also kept a backup
+        // in _addDateFiltersToVectorLayers() we rewrote the layers' filters to support date filtering, but we also kept a backup
         // restore that backup now, so the layers are back where they started
         const vecmap = this.getRealGlMap();
         this._getFilteredvectorLayers().forEach((layer) => {
             vecmap.setFilter(layer.id, layer.oldfiltersbackup);
         });
     },
     _applyDateFilterToLayers: function () {
-        // back in _setupDateFiltersForLayers() we prepended a filtering clause as filters[1] which filters for "eternal" features lacking a OSM ID
+        // back in _addDateFiltersToVectorLayers() we prepended a filtering clause as filters[1] which filters for "eternal" features lacking a OSM ID
         // here in _applyDateFilterToLayers() we add a second part to that, for features with a start_date and end_date fitting our date
+        // the new date filtering expression is:
+        // numerical start date either absent (beginning/end of time) or else within range, and same for numerical end date
         const deccurrent = this.getDate(true);
         const datesubfilter = [
             'all',
-            // numerical start/end date either absent (beginning/end of time) or else within range
-            [ 'any', ['!has', 'start_decdate'], ['<=', 'start_decdate', deccurrent] ],
-            [ 'any', ['!has', 'end_decdate'], ['>=', 'end_decdate', deccurrent] ],
+            [
+                'any',
+                ['!', ['has', 'start_decdate']],
+                ['<=', ['to-number', ['get', 'start_decdate'], -Infinity], ['literal', deccurrent]]
+            ],
+            [
+                'any',
+                ['!', ['has', 'end_decdate']],
+                ['>=', ['to-number', ['get', 'end_decdate'], Infinity], ['literal', deccurrent]]
```

Comment:
As far as I can tell, `deccurrent` is guaranteed to be a number, so there’s no need to wrap it in a `literal` expression. We only need to wrap arrays and objects in `literal` expressions.

Reply:
Experimentally, I can say that the `literal` was required in order for it to work reliably. I'm not sure why since Infinity and plain numbers often do work here. But I can say that it definitely did not work when that was not used.